vagrant/provisioning/roles/alfresco-site/tasks/main.yml

```diff
@@ -0,0 +1,210 @@
+  file:
+    path: cookie.txt
+    state: absent
+  changed_when: false
+    
+# note, can't use get_uri here since we need to keep the cookie jar
+- name: login to Share
+  command: curl -k -sS --user admin:admin --cookie cookie.txt --cookie-jar cookie.txt -H "Content-Type:application/x-www-form-urlencoded" --data "success=/share/page/" --data "failure=/share/page/?error=true" --data "username=admin" --data "password=admin" -X POST "{{ alfresco_url }}/share/page/dologin"
+  args:
+    warn: false
+  changed_when: false
+
+- name: check if login succeeded
+  command: cat cookie.txt
+  register: cookie_out
+  failed_when: '"alfLogin" not in cookie_out.stdout'
+  changed_when: false
+
+- name: get the CSRFToken
+  command: curl -k -sS --user admin:admin --cookie cookie.txt --cookie-jar cookie.txt "{{ alfresco_url }}/share/service/modules/authenticated?a=user"
+  args:
+    warn: false
+  changed_when: false
+
+- name: check if we got the CSRFToken
+  command: cat cookie.txt
+  register: csrf_cookie_out
+  failed_when: '"Alfresco-CSRFToken" not in csrf_cookie_out.stdout'
+  changed_when: false
+
+- name: parse the CSRFToken
+  shell: egrep ^{{ alfresco_host }} cookie.txt | grep Alfresco-CSRFToken | awk '{ print $7 }'
+  register: csrftoken_parsed
+  changed_when: false
+
+- name: decode csrftoken
+  set_fact:
+    csrftoken: "{{ csrftoken_parsed.stdout|urldecode }}"
+
+- name: create acm share site
+  include_tasks: create-site.yml
+  vars:
+    s_title: "acm"
+    s_name: "acm"
+    s_description: "ArkCase Collaboration Site"
+    s_preset: "site-dashboard"
+    s_type: "{http://www.alfresco.org/model/site/1.0}site"
+    s_compliance: "{http://www.alfresco.org/model/recordsmanagement/1.0}rmsite"
+    s_csrftoken: "{{ csrftoken }}"
+
+- name: list the document library, so we can then add some folders
+  uri:
+    url: "{{ alfresco_url }}/alfresco/s/slingshot/doclib/container/acm/documentlibrary"
+    user: admin
+    password: admin
+    validate_certs: no
+    return_content: yes
+  register: doclib_out
+  failed_when: '"nodeRef" not in doclib_out.content'
+
+- name: see what folders already exist
+  uri:
+    url: "{{ alfresco_url }}/alfresco/s/slingshot/doclib/treenode/site/acm/documentLibrary?perms=false&children=false&max=1000"
+    user: admin
+    password: admin
+    validate_certs: no
+    return_content: yes
+  register: existing_folders_out
+
+- name: ensure the right folders are in the document library
+  uri:
+    url: "{{ alfresco_url }}/alfresco/s/api/site/folder/acm/documentlibrary"
+    method: POST
+    user: admin
+    password: admin
+    validate_certs: no
+    return_content: yes
+    body_format: json
+    body:
+      name: "{{ item }}"
+      title: "{{ item }}"
+  register: folder_create_out
+  failed_when: '"site" not in folder_create_out.content'
+  loop: "{{ content_management_folders }}"
+  when: 'item not in existing_folders_out.content'
+
+- name: create records management share site
```

Comment:
We can introduce a variable not to install the RM site, when clients don't want it.

Reply:
@bojmil-armedia we have 'alfresco_install_governance_services' in the facts file already ... but in the alfresco role we may or may not check it.